src/doc/rustc/src/platform-support/wasm32-wasi-preview1-threads.md

```diff
@@ -0,0 +1,151 @@
+   will cause Rust code to be linked against the libc.a that the specified
+   `clang` provides.
+3. If you're building a staticlib and integrating Rust code elsewhere, then
+   compiling with `-C target-feature=-crt-static` is all you need to do.
+
+All in all, by default, no external dependencies are required. You can
+compile `wasm32-wasi-preview1-threads` binaries straight out of the box. You can't, however,
+reliably interoperate with C code in this mode (yet).
+
+
+This target is not a stable target. This means that there are not many engines
+which implement the `wasi-threads` feature and if they do they're likely behind a
+flag, for example:
+
+* Wasmtime - `--wasm-features=threads --wasi-modules=experimental-wasi-threads`
+
+Also note that at this time the `wasm32-wasi-preview1-threads` target assumes the
+presence of other merged wasm proposals such as (with their LLVM feature flags):
+
+* [Bulk memory] - `+bulk-memory`
+* Mutable imported globals - `+mutable-globals`
+* Atomics - `+atomics`
+
+[Bulk memory]: https://github.com/WebAssembly/spec/blob/main/proposals/bulk-memory-operations/Overview.md
+
+LLVM 16 is required for this target. The reason is related to linker flags: prior to LLVM 16, --import-memory and --export-memory were not allowed together. The reason both are needed is an artifact of how WASI currently does things; see https://github.com/WebAssembly/WASI/issues/502 for more details.
+
+The `wasm32-wasi-preview1-threads` target intends to match the default Clang targets
+for its `"C"` ABI, which is likely to be the same as Clang's
+`wasm32-wasi-preview1` largely.
+
+> **Note**: due to the relatively early-days nature of this target when working
+> with this target you may encounter LLVM bugs. If an assertion hit or a bug is
+> found it's recommended to open an issue either with rust-lang/rust or ideally
+> with LLVM itself.
+
+## Building the target
+
+You can build Rust with support for the target by adding it to the `target`
+list in `config.toml`, and the target also requires `wasi-root` to be built to work.
+
+```toml
+[build]
+target = ["wasm32-wasi-preview1-threads"]
+
+[target.wasm32-wasi-preview1-threads]
+wasi-root = ".../wasi-libc/sysroot"
+```
+
+Build the stage 1 rustc compiler: `./x.py build --stage 1 --target wasm32-wasi-preview1-threads`
+
+## Building Rust programs
+
+If `rustc` has support for that target.
+```shell
+$ rustc -v --target=wasm32-wasi-threads threads.rs
+$ ls threads.wasm
+```
+
+If `rustc` doesn't have support for this target you will need to build Rust with
+the target enabled (see"Building the target" above). Ensure that `rust-lld` can be found
+in the `$PATH` environment variable. If it's not there you most likely can find it in
+`$HOME/.rustup/...` and update the `$PATH` variable. If it doesn't work you might need to
+run `rustup update` so the linker has all required flags.
+
+```shell
+$ rustc +stage1 -v --target=wasm32-wasi-threads threads.rs
+$ ls threads.wasm
+```
```

Comment:
When describing building for a target in platform-support.md, it's totally fine to assume they are using the "complete" platform toolchain and thus suggest that, like most Rust developers, they might want to simply run
```shell
cargo +nightly build --target=wasm32-wasi-preview1-threads -Zbuild-std
```

It should work correctly if whatever additional configuration needed is included in the environment. If you can't get that working, please file a bug [over here](https://github.com/rust-lang/wg-cargo-std-aware/issues)? As this is much simpler than "rebuild rustc", you might want to put it first.

Reply:
Can I check manually somehow whether it's going to work? I think there might be problems because as I mentioned locally it's needed to have wasi-root set in `config.toml` and I don't understand how to set it in the way you suggested

Reply:
If there's no way to pass it via flag, environment variable, or linker arguments, then please file an issue describing that.

Reply:
Would it be fine if I just update my PR so it gets inherited from an environment variable that I'll introduce?

Reply:
Seems like it should be possible to create `.cargo/config.toml` in project directory https://doc.rust-lang.org/cargo/reference/config.html or pass it through `--config` https://doc.rust-lang.org/cargo/reference/config.html#command-line-overrides but should I be able to test it before merging the change since it's in`stage1` that has been already built locally and not used/built from `nightly`?

Reply:
Oh sure, if that works, it sounds fine as a shim for now!